File: src/main/java/com/freya02/botcommands/internal/application/slash/autocomplete/suppliers/ChoiceSupplierChoices.java
package com.freya02.botcommands.internal.application.slash.autocomplete.suppliers;

import com.freya02.botcommands.internal.application.slash.SlashCommandInfo;
import com.freya02.botcommands.internal.application.slash.autocomplete.AutocompletionHandlerInfo;
import com.freya02.botcommands.internal.application.slash.autocomplete.ChoiceSupplier;
import net.dv8tion.jda.api.events.interaction.command.CommandAutoCompleteInteractionEvent;
import net.dv8tion.jda.api.interactions.commands.Command;

import java.util.Collection;
import java.util.List;

public class ChoiceSupplierChoices implements ChoiceSupplier {
	private final AutocompletionHandlerInfo handlerInfo;

	public ChoiceSupplierChoices(AutocompletionHandlerInfo handlerInfo) {
		this.handlerInfo = handlerInfo;
	}

	@SuppressWarnings("unchecked")
	@Override
	public List<Command.Choice> apply(SlashCommandInfo slashCommand, CommandAutoCompleteInteractionEvent event, Collection<?> collection) throws Exception {
		final Collection<Command.Choice> choices = (Collection<Command.Choice>) collection;

		return choices.stream()
				.limit(handlerInfo.getMaxChoices())
				.toList();
	}
}

File: src/main/kotlin/com/freya02/botcommands/internal/commands/application/slash/autocomplete/suppliers/ChoiceSupplierChoices.kt
package com.freya02.botcommands.internal.commands.application.slash.autocomplete.suppliers

import net.dv8tion.jda.api.events.interaction.command.CommandAutoCompleteInteractionEvent
import net.dv8tion.jda.api.interactions.commands.Command

internal class ChoiceSupplierChoices(private val numChoices: Int) : ChoiceSupplier {
    @Suppress("UNCHECKED_CAST")
    override fun apply(event: CommandAutoCompleteInteractionEvent, collection: Collection<*>): List<Command.Choice> {
        return collection.take(numChoices) as List<Command.Choice>
    }
}